pennylane/transforms/qfunc_transforms.py

```diff
@@ -182,10 +182,10 @@ def internal_wrapper(*args, **kwargs):
         tape = make_qscript(fn)(*args, **kwargs)
         tape = tape_transform(tape, *transform_args, **transform_kwargs)
 
-        if len(tape.measurements) == 1:
-            return tape.measurements[0]
-
-        return tape.measurements
+        num_measurements = len(tape.measurements)
+        if num_measurements == 0:
+            return None
+        return tape.measurements[0] if num_measurements == 1 else tape.measurements

```

Comment:
We should have a unit test for this behaviour too.

Reply:
added